acceptance/tests/pxp_module_bolt_command/run_command.rb

```diff
@@ -0,0 +1,23 @@
+require 'pxp-agent/bolt_pxp_module_helper.rb'
+
+test_name 'run_command task' do
+  step 'Ensure each agent host has pxp-agent running and associated' do
+    agents.each do |agent|
+      on agent, puppet('resource service pxp-agent ensure=stopped')
+      create_remote_file(agent, pxp_agent_config_file(agent), pxp_config_hocon_using_puppet_certs(master, agent))
+      on agent, puppet('resource service pxp-agent ensure=running')
+
+      assert(is_associated?(master, "pcp://#{agent}/agent"),
+             "Agent #{agent} with PCP identity pcp://#{agent}/agent should be associated with pcp-broker")
+    end
+  end
+
+  step 'Run a command (`echo`) on agent hosts' do
+    agents.each do |agent|
+      cmd = agent.platform =~ /windows/ ? 'cmd /c echo hello' : 'echo hello'
```

Comment:
My recollection is that pxp-agent has always invoked the executable directly rather than running in the context of a shell. Bolt over other transports has different semantics. I wouldn't change it in this PR, but do we want to rethink how `run_command` works with pxp-agent to more closely mirror Bolt with other transports?

Reply:
Based on investigation around shell splitting, I'm no longer certain that's the case.

Reply:
I don't have enough experience within this project to have much of an opinion, but @branan ^ this seems worth discussing sooner rather than later

Reply:
@MikaelSmith the use of cmd.exe in this acceptance test isn't because we are running in the shell for run_command. There just isn't an echo executable on windows, so there was no simple way to create simple stdout without executing in cmd.exe first.

The actual invocation of the command to be used should be using leatherman::execute

Reply:
Oh, wait, I guess maybe what your suggesting is we _should_ be executing in the shell?

Reply:
The question was just inspired by the example. I think it's confusing to folks when the answer to that question changes depending on what transport they're using, but I might be confusing this with the Docker transport (where it can't assume a shell because there might not be a shell executable).

Reply:
If we add some unsuccessful command invokes at the unit level, we can probably get away with doing happy path testing at the acceptance level. Otherwise, we should consider some unhappy path tests here (failed and pxp_errored)? As we're not currently mucking with the protocol, we don't need to explore that space.